examples/handshake.rs

```diff
@@ -0,0 +1,102 @@
+use rand::Rng;
+use std::io::Write;
+use std::net::{IpAddr, Ipv4Addr, Shutdown, SocketAddr, TcpStream};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use bitcoin::consensus::encode;
+use bitcoin::network::address;
+use bitcoin::network::constants;
+use bitcoin::network::message;
+use bitcoin::network::message_network;
+use bitcoin::network::stream_reader::StreamReader;
+
+fn main() {
+    // This example establishes a connection to a Bitcoin node, sends the intial
+    // "version" message, waits for the reply, and finally closes the connection.
+
+    let address = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(130, 149, 80, 221)), 8333);
+
+    let version_message = build_version_message(address.clone());
```

Comment:
`ScoketAddr` is Copy, no need to clone https://doc.rust-lang.org/std/net/enum.SocketAddr.html#impl-Copy

Reply:
Ah, thanks, you are right!

Reply:
Tip: use `cargo clippy`, or even better - RLS with clippy support turned on.

Reply:
I usually do, however in this case `cargo clippy` yields a lot of warnings and even errors for the rust-bitcoin codebase. Not sure if fixing all of these is a goal, if so, I could have a look at it in another PR.
 
Apart from that, checking out the RLS is something I intended to do for some time, thanks for the reminder! :+1:

Reply:
@tnull See https://github.com/rust-bitcoin/rust-bitcoin/pull/308